### PR TITLE
Direct object passing for decimator (#1591)

### LINKED ISSUE #1591: Support direct value in the chart constructor in the Page Builder API
### Description

If you want to set a decimator in a chart created with the Page Builder API, you must create an intermediate variable whose name is set to the `decimator` property:
```
the_decimator = MinMaxDecimator(...)
...
tgb.chart("{data}", ..., decimator="{the_decimator}")
```

### Solution Proposed

A far natural (and shorter) syntax would be:
```
tgb.chart("{data}", ..., decimator=MinMaxDecimator(...))
```

### Impact of Solution

Simplify the use proposing natural Python constructs.

### Additional Context

_No response_

### Acceptance Criteria

- [ ] Ensure new code is unit tested, and check code coverage is at least 90%.
- [ ] Create related issue in taipy-doc for documentation and Release Notes.
- [ ] Check if a new demo could be provided based on this, or if legacy demos could be benefit from it.
- [ ] Ensure any change is well documented.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by dinhlongviolin1**
Resolve #1591 

The idea behind this is that a variable would be embedded in the caller frame -> we can use the embedded object for the decimator. This could be extended to all objects or any necessary objects in the future.

@FredLL-Avaiga I did some refactoring on the lambda so it is a bit cleaner to read

### CODE REVIEW

**Review by FredLL-Avaiga — APPROVED** [review]
I like it

File: taipy/gui/builder/_element.py [hunk 1]
```diff
@@ -93,47 +95,66 @@ def _parse_property(self, key: str, value: t.Any) -> t.Any:
...
-                    self._lambdas[lambda_name] = lambda_text
-                    return f'{{{lambda_name}({", ".join(args)})}}'
-            except Exception as e:
-                _warn("Error in lambda expression", e)
+                return value if value.__name__.startswith("<") else value.__name__
+            # Parse lambda function
+            if (lambda_name := self.__parse_lambda_property(key, value)) is not None:
+                return lambda_name
+        # Embed value in the caller frame
+        if isinstance(value, Decimator) and "decimator" in key:
```
Comment (FredLL-Avaiga) [inline taipy/gui/builder/_element.py]: this could be `not instance(value, str) and "decimator" in key` => no `Decimator` dependency

**Review by FredLL-Avaiga — DISMISSED** [review]

File: taipy/gui/builder/_element.py (hunk 1 above)
Comment (FabienLelaquais) [reply taipy/gui/builder/_element.py]: Indeed that would make things far more extensible in the future - on top of avoiding this annoying dependency.

**Review by FabienLelaquais — DISMISSED** [review]
Neat!!

File: taipy/gui/builder/_element.py [hunk 2]
```diff
@@ -93,47 +94,66 @@ def _parse_property(self, key: str, value: t.Any) -> t.Any:
...
-                    self._lambdas[lambda_name] = lambda_text
-                    return f'{{{lambda_name}({", ".join(args)})}}'
-            except Exception as e:
-                _warn("Error in lambda expression", e)
+                return value if value.__name__.startswith("<") else value.__name__
+            # Parse lambda function
+            if (lambda_name := self.__parse_lambda_property(key, value)) is not None:
+                return lambda_name
+        # Embed value in the caller frame
+        if not isinstance(value, str) and "decimator" in key:
```
Comment (FabienLelaquais) [inline taipy/gui/builder/_element.py]: May be a little comment on that specific property name test?
Like... "Properties that hold backend-only object".
So far, only 'decimator' shows up, but that may change in the future don't you think?

File: taipy/gui/builder/_element.py [hunk 3]
```diff
@@ -93,47 +94,66 @@ def _parse_property(self, key: str, value: t.Any) -> t.Any:
...
+        Return the Taipy expression of the embedded object
+        """
+        frame_locals = self.__calling_frame.f_locals
+        obj_var_name = self._TAIPY_EMBEDDED_PREFIX + obj.__class__.__name__
+        index = 0
+        while f"{obj_var_name}_{index}" in frame_locals:
+            index += 1
+        obj_var_name = f"{obj_var_name}_{index}"
+        frame_locals[obj_var_name] = obj
+        return f"{{{obj_var_name}}}" if is_expression else obj_var_name
```
Comment (FabienLelaquais) [inline taipy/gui/builder/_element.py:156]: 👍

**Review by FabienLelaquais — DISMISSED** [review]
😮looks good!

File: taipy/gui/builder/_element.py (hunk 2 above)
Comment (dinhlongviolin1) [reply taipy/gui/builder/_element.py]: i will add a global list for these properties

**Review by FredLL-Avaiga — APPROVED** [review]

**Pull request merged**